lib/bitcoin/BitcoinClient.ts

```diff
@@ -108,34 +171,23 @@ export default class BitcoinClient implements IBitcoinClient {
     const hexEncodedTransaction = await this.rpcCall(request, true);
     const transactionBuffer = Buffer.from(hexEncodedTransaction, 'hex');
 
-    return BitcoinClient.createTransactionFromBuffer(transactionBuffer);
+    const bcoreTransaction = BitcoinClient.createBcoreTransactionFromBuffer(transactionBuffer);
+
+    return BitcoinClient.createBitcoinTransactionModel(bcoreTransaction);
   }
 
-  public async getUnspentCoins (address: Address): Promise<Transaction.UnspentOutput[]> {
+  public async getUnspentCoins (): Promise<BitcoinUnspentCoinsModel[]> {
 
-    // Retrieve all transactions by addressToSearch via BCoin Node API /tx/address/$address endpoint
-    const addressToSearch = address.toString();
-    console.info(`Getting unspent coins for ${addressToSearch}`);
-    const request = {
-      method: 'listunspent',
-      params: [
-        null,
-        null,
-        [addressToSearch]
-      ]
-    };
-    const response: Array<any> = await this.rpcCall(request, true);
+    const unspentOutputs = await this.getUnspentOutputs(this.privateKeyAddress);
 
-    const unspentTransactions = response.map((coin) => {
-      return new Transaction.UnspentOutput(coin);
+    const unspentTransactions = unspentOutputs.map((unspentOutput) => {
+      return { satoshis: unspentOutput.satoshis };
     });
 
-    console.info(`Returning ${unspentTransactions.length} coins`);
-
     return unspentTransactions;
   }
 
-  public async importPublicKey (publicKeyAsHex: string, rescan: boolean): Promise<void> {
+  private async importPublicKey (publicKeyAsHex: string, rescan: boolean): Promise<void> {

```

Comment:
Rename to `addWatchOnlyAddressToWallet()`

For reference see https://bitcoincore.org/en/doc/0.18.0/rpc/wallet/importpubkey/